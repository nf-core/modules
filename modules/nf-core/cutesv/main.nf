process CUTESV {
    tag "$meta.id"
    label 'process_high'

    conda "bioconda::cutesv=2.0.2"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/cutesv:1.0.12--pyhdfd78af_0' :
        'quay.io/biocontainers/cutesv:2.0.2--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(bam), path(bai)
    tuple val(meta2), path(fasta)

    output:
    tuple val(meta), path("*.vcf"), emit: vcf
    path "versions.yml"                  , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    cuteSV \
        ${bam} \\
        ${fasta} \\
        ${prefix}.vcf \\
        . \\
        --threads $task.cpus \\
        $args
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cuteSV: \$( cuteSV --version 2>&1 | sed 's/cuteSV //g' )
    END_VERSIONS
    """
}
