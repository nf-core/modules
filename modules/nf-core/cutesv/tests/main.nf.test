nextflow_process {

    name "Test Process CUTESV"
    
    script "../main.nf"
    process "CUTESV"
   
    tag "modules"
    tag "modules_nfcore"
    tag "cutesv"
   
    test("sarscov2 - bam - fasta") {
        when {
            
            process {
                """
                input[0] = Channel.of([[id:'test'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/bam/test.sorted.bam',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/bam/test.sorted.bam.bai',checkIfExists:true),
                ])
                input[1] = Channel.of([[id:'fasta'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta',checkIfExists:true)
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.vcf.collect{ [it[0], path(it[1]).vcf.summary] }
                ).match() }
            )
        }
    }

    test("sarscov2 - bam - fasta - stub") {

        options '-stub'

        when {
            
            process {
                """
                input[0] = Channel.of([[id:'test'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/bam/test.sorted.bam',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/bam/test.sorted.bam.bai',checkIfExists:true),
                ])
                input[1] = Channel.of([[id:'fasta'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta',checkIfExists:true)
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    process.out.versions.collect{ path(it).yaml }
                ).match() }
            )
        }
    }
}