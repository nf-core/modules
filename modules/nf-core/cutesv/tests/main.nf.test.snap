{
    "sarscov2 - bam - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,9d8a5229ecda416f1a538ac0fa22051c"
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,9d8a5229ecda416f1a538ac0fa22051c"
                ]
            },
            [
                {
                    "CUTESV": {
                        "cuteSV": "2.0.2"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T17:39:59.960829783"
    },
    "sarscov2 - bam - fasta": {
        "content": [
            [
                "versions.yml:md5,9d8a5229ecda416f1a538ac0fa22051c"
            ],
            [
                [
                    {
                        "id": "test"
                    },
                    "VcfFile [chromosomes=[], sampleCount=1, variantCount=0, phased=true, phasedAutodetect=true]"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-25T17:39:53.79202941"
    }
}